### ISSUE TITLE
When I update `torchaudio` from 0.9.0 to 0.10.0, it raise an error `formats: amr-nb can't encode AMR-NB to 16-bit`.

### ISSUE THREAD

**Issue by TeaPoly** [pytorch/audio#2099]
### 🐛 Describe the bug

When I update `torchaudio` from 0.9.0 to 0.10.0, it raise an error `formats: amr-nb can't encode AMR-NB to 16-bit` for following code:

``` python
import numpy as np
import soundfile

import torch
import torchaudio

path = "test.wav"

# test.wav is 8kHz/16bit PCM wav audio
samples, sample_rate = soundfile.read(path, dtype='float32')

speech = torchaudio.functional.apply_codec(
    torch.from_numpy(np.expand_dims(samples, axis=0)),
    sample_rate= sample_rate,
    format="amr-nb",
    compression=5)
```

Is it means` amr-nb` is not support 16-bit/sample width audio now?

### Versions

Collecting environment information...
```
PyTorch version: 1.10.0
Is debug build: False
CUDA used to build PyTorch: 11.1
ROCM used to build PyTorch: N/A

OS: Ubuntu 18.04.3 LTS (x86_64)
GCC version: (Ubuntu 5.5.0-12ubuntu1) 5.5.0 20171010
Clang version: 6.0.0-1ubuntu2 (tags/RELEASE_600/final)
CMake version: version 3.10.2
Libc version: glibc-2.27

Python version: 3.8.12 (default, Oct 12 2021, 13:49:34) [GCC 7.5.0] (64-bit runtime)
Python platform: Linux-4.15.0-65-generic-x86_64-with-glibc2.17
Is CUDA available: True
CUDA runtime version: Could not collect
GPU models and configuration:
GPU 0: NVIDIA GeForce GTX 1080 Ti
GPU 1: NVIDIA GeForce GTX 1080 Ti
GPU 2: NVIDIA GeForce GTX 1080 Ti

Nvidia driver version: 465.19.01
cuDNN version: Probably one of the following:
/usr/lib/x86_64-linux-gnu/libcudnn.so.7.6.4
/usr/local/cuda-10.0/targets/x86_64-linux/lib/libcudnn.so.7.6.4
HIP runtime version: N/A
MIOpen runtime version: N/A

Versions of relevant libraries:
[pip3] numpy==1.20.3
[pip3] torch==1.10.0
[pip3] torchaudio==0.10.0
[pip3] torchvision==0.11.1
[conda] blas 1.0 mkl
[conda] cudatoolkit 11.1.1 h6406543_8 conda-forge
[conda] ffmpeg 4.3 hf484d3e_0 pytorch
[conda] mkl 2021.4.0 h06a4308_640
[conda] mkl-service 2.4.0 py38h497a2fe_0 conda-forge
[conda] mkl_fft 1.3.1 py38hd3c417c_0
[conda] mkl_random 1.2.2 py38h1abd341_0 conda-forge
[conda] numpy 1.20.3 pypi_0 pypi
[conda] pytorch 1.10.0 py3.8_cuda11.1_cudnn8.0.5_0 pytorch
[conda] pytorch-mutex 1.0 cuda pytorch
[conda] torchaudio 0.10.0 py38_cu111 pytorch
[conda] torchvision 0.11.1 py38_cu111 pytorch
```

**Comment by mthrok** [comment]
Hi @TeaPoly 

tl;dr It's just warning and you can ignore it. Please refer to the figure bellow for the resulting signal.

**Note** <ins>Even if the source audio is 16 bits, when it is loaded with `soundfile.read(dtype='float32')`, the data are represented with 32-bit floating point value. So the fact that source audio being 16 bits is not relevant.</ins>

The warning is issued because torchaudio's internal implementation passes 16bits to underlying amr-nb encoder . [[code](https://github.com/pytorch/audio/blob/c6de2a194bcfa4d0a50fa2058d195851b8aa1a47/torchaudio/csrc/sox/utils.cpp#L320-L326)].
sox issues the warning, because `amr-nb` cannot deal with exact 16 bit because `amr-nb` format has fixed sampling rate of 8000 and fixed bit rate determined by `compression` parameter..

> AMR-NB audio has a fixed sampling rate of 8 kHz and supports encoding to the following bit-rates (as selected by the −C option): 0 = 4.75 kbit/s, 1 = 5.15 kbit/s, 2 = 5.9 kbit/s, 3 = 6.7 kbit/s, 4 = 7.4 kbit/s 5 = 7.95 kbit/s, 6 = 10.2 kbit/s, 7 = 12.2 kbit/s.
http://sox.sourceforge.net/soxformat.html

Since it's kinda hard to save the augmented audio without introducing another effect, the best way to check the effect is to visualize it. The following script generates sine wave of `freq=300` Hz with sampling rate of 8k Hz, and applies `amr-nb` codec.

The resulting spectrogram preserve intensity around 300 Hz and the waveform is disturbed.


```python
import torch
import torchaudio

import matplotlib.pyplot as plt

sample_rate = 8000
num_frames = 2000
freq = 300

theta = torch.linspace(0, 2 * 3.14 * freq * num_frames / sample_rate, num_frames)
audio = torch.sin(theta)[None, ...]

augmented = torchaudio.functional.apply_codec(
    audio, sample_rate=sample_rate, format="amr-nb", compression=5)

fig, axs = plt.subplots(2, 2)
axs[0][0].specgram(audio[0], Fs=sample_rate)
axs[1][0].specgram(augmented[0], Fs=sample_rate)
axs[0][1].plot(audio[0])
axs[1][1].plot(augmented[0])
plt.show()
```
<img width="1215" alt="Screen Shot 2021-12-23 at 10 05 32 PM" src="https://user-images.githubusercontent.com/855818/147311032-9cbfbbda-61cb-4272-a8cd-c4c7f92e0be7.png">

**Comment by TeaPoly** [comment]
Thank you for your patient answer.

**Comment by zylo117** [comment]
I found out this issue exists on mp3 and ogg too. This warning is so annoying, Can we remove it?

**Comment by jyn282** [comment]
@zylo117  I'm so annoyed by that warning, too.  Do you find a way to turn off that warning?
